### PR TITLE
#15 BaseIcons

### PR DESCRIPTION
**Pull request by adrianjost**
# Issue #15 

Big Thanks to CodinCat ( https://stackoverflow.com/questions/44643340/code-splitting-import-breaks-jest-tests ) for sharing a fix for the dynamic import bug that caused jest coverage crashing.

Simply replacing the package `@babel/plugin-syntax-dynamic-import` with `babel-plugin-dynamic-import-node` fixed it.

## Description

- place any svg file at `@assets/icons` and then use them with `<BaseIcon source="custom" icon="fileName" />`
- replaced existing icon imports with new BaseIcon
- switched from FontAwesome to Material Icons cause they were easier to implement. Buggy Version of FontAwesome is still included
- Tests will be added in a seperate branch after #81 is merged

## Screenshot
image

### CODE REVIEW

**Comment by janrenz** [comment]
the line height (baseline) of the icons seems to jump ...

**Comment by adrianjost** [comment]
@janrenz that right. We need to modify the existing svg files. They currently have some inconsistently large whitespace around them. After editing those icons it should be equal.

File: src/components/TemplateCourseWizard.vue [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	<div class="root">
 		<div class="header">
 			<div class="header-icon">
-				<SpaceShuttle />
+				<BaseIcon source="custom" icon="shuttle" />
```
[the remark this answers is not in the document]
Comment (dominikbruechner) [reply src/components/TemplateCourseWizard.vue]: approved but keep in mind fixing it somewhere

File: src/components/ui/BaseIcon.vue [hunk 2]
```diff
@@ -1,47 +1,70 @@
...
+	methods: {
+		loadIcon() {
+			if (this.source === "custom") {
+				// @assets/icons
+				return import(`@assets/icons/${this.icon}.svg`).then((iconPath) => {
+					this.svgPath = iconPath.default;
+				});
+			}
+			if (this.source === "material") {
+				// https://material.io/tools/icons/?style=baseline
```
Comment (dominikbruechner) [inline src/components/ui/BaseIcon.vue]: do we still need this comment?

File: src/components/TemplateCourseWizard.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (adrianjost) [reply src/components/TemplateCourseWizard.vue]: You can use font-size: 60px on the BaseIcon to resize it to 60px. Just like we would do it with font awesome

File: src/components/ui/BaseIcon.vue (hunk 2 above)
Comment (adrianjost) [reply src/components/ui/BaseIcon.vue]: I think it's handy for quickly finding the available Icons. But I'm fine with removing it

File: src/components/TemplateCourseWizard.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (adrianjost) [reply src/components/TemplateCourseWizard.vue]: image

**Pull request merged**